contributors/guide/pull-requests.md

```diff
@@ -302,34 +313,228 @@ For more information, see [squash commits](./github-workflow.md#squash-commits).
 
  Don't squash when there are independent changes layered to achieve a single goal. For instance, writing a code munger could be one commit, applying it could be another, and adding a precommit check could be a third. One could argue they should be separate pull requests, but there's really no way to test/review the munger without seeing it applied, and there needs to be a precommit check to ensure the munged output doesn't immediately get out of date.
 
-A commit, as much as possible, should be a single logical change.
+## 7. Commit Message Guidelines
 
-## 7. KISS, YAGNI, MVP, etc.
+PR comments are not represented in the commit history. Commits and their commit
+messages are the _"permanent record"_ of the changes being done in your PR and
+their commit messages should accurately describe both _what_ and _why_ it is
+being done.
 
-Sometimes we need to remind each other of core tenets of software design - Keep It Simple, You Aren't Gonna Need It, Minimum Viable Product, and so on. Adding a feature "because we might need it later" is antithetical to software that ships. Add the things you need NOW and (ideally) leave room for things you might need later - but don't implement them now.
+Commit messages are comprised of two parts; the subject and the body.
 
-## 8. It's OK to Push Back
+The subject is the first line of the commit message and is often the only part
+of the commit message needed for small changes. These are often done as "one
```

Comment:
Could this be more clear?  Eg:

...of the commit message needed for the smallest of trivial changes.  Those may be done as "one liners" with the git `-m` or `--message` flag, but only if the _what_ and especially _why_ can be fully described in that few words.  Typing a few further sentences of clarification is a useful investment in time both for your reviews and overall later project maintenance.

Reply:
Also, in keeping with the theme of GitHub PR artifacts are not in the git commit history:  We have useful area labels for organizing on GitHub.  Recommending using those, or similar descriptive lead in words, in the summary line would further give that useful organizing to those doing project/code maintenance relative to what you see in the git history.

Reply:
Referenced our kind and area labels (with links) for commit message prefixes.